polars/polars-time/src/chunkedarray/utf8/mod.rs

```diff
@@ -275,29 +277,56 @@ pub trait Utf8Methods: AsUtf8 {
-                            Ok(nd) => return Some(nd),
-                            Err(e) => {
-                                let e: ParseErrorByteCopy = e.into();
-                                match e.0 {
-                                    ParseErrorKind::TooLong => {
-                                        s = &s[..s.len() - 1];
-                                    }
-                                    _ => {
-                                        s = &s[i..];
+                        match tz_aware {
+                            true => {
+                                match DateTime::parse_from_str(s, fmt)
+                                    .map(|dt| func(dt.naive_utc()))
+                                {
+                                    Ok(nd) => return Some(nd),
+                                    Err(e) => {
+                                        let e: ParseErrorByteCopy = e.into();
+                                        match e.0 {
+                                            ParseErrorKind::TooLong => {
+                                                s = &s[..s.len() - 1];
+                                            }
+                                            _ => {
+                                                s = &s[i..];
+                                            }
+                                        }
                                     }
                                 }
                             }
+                            false => match NaiveDateTime::parse_from_str(s, fmt).map(func) {
+                                Ok(nd) => return Some(nd),
+                                Err(e) => {
+                                    let e: ParseErrorByteCopy = e.into();
+                                    match e.0 {
+                                        ParseErrorKind::TooLong => {
+                                            s = &s[..s.len() - 1];
+                                        }
+                                        _ => {
+                                            s = &s[i..];
+                                        }
+                                    }
+                                }
+                            },
                         }
                     }
                     None
                 }
             })
             .collect_trusted();
         ca.rename(utf8_ca.name());
-        match tz {
+        match (tz_aware, tz) {
             #[cfg(feature = "timezones")]
-            Some(tz) => ca.into_datetime(tu, None).replace_time_zone(Some(tz), None),
+            (false, Some(tz)) => ca.into_datetime(tu, None).replace_time_zone(Some(tz), None),
+            #[cfg(feature = "timezones")]
+            (true, None) => Ok(ca.into_datetime(tu, Some("UTC".to_string()))),
+            (true, Some(_)) => {
+                // Can't use tz-aware format with tz-aware dtype, this would already have
+                // errored earlier.
```

Comment:
note to self: this should probably be allowed, but only if the dtype has time zone UTC - one for a separate PR anyway